Sources/Commands/SwiftTestTool.swift

```diff
@@ -188,6 +199,86 @@ public struct SwiftTestTool {
         let result: Void? = try? system(args, environment: NSProcessInfo.processInfo().environment)
         return result != nil
     }
+
+    /// Runs the corresponding tool to get the tests JSON.
+    /// On OSX, we use the swiftpm-xctest-helper tool bundled with swiftpm.
+    /// On Linux, XCTest can dump the json using `--dump-tests-json` mode.
+    ///
+    /// - Parameters:
+    ///     - path: Path to the XCTest bundle(OSX) or executable(Linux).
+    ///     - tempDirectory: Path to a directory where a temporary output
+    ///                      file will be created containing the JSON.
+    ///                      The File is deleted when this method returns.
+    private func listTests(path: String, tempDirectory: String) throws {
```

Comment:
I would expect this function to just be infrastructure which returns the test listing, for use in other infrastructure, and the actual listing functionality would be in the client.

I also don't think it should need the `tempDirectory` argument, that is an implementation detail. It can just create the temp file and throw it away when done.